sh/tests/file-checker_test.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS='
+'
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
```

Comment:
Suggestion:
at the start you can create the folder and files
```
if test ! -e file-checker; then
    mkdir file-checker
    cd file-checker
    touch readable-and-writable readable-only readable-and-executable readable-writable-executable
    chmod -x "$script_dirS/file-checker/readable-and-writable"
    chmod -xw "$script_dirS/file-checker/readable-only"
    chmod -w "$script_dirS/file-checker/readable-and-executable"
    chmod +x "$script_dirS/file-checker/readable-and-executable"
    chmod +x "$script_dirS/file-checker/readable-writable-executable"
    cd ..
fi
```
At the end after you test it you can remove it:

```
challenge "$script_dirS/file-checker/readable-only"
challenge "$script_dirS/file-checker/readable-and-writable"
challenge "$script_dirS/file-checker/readable-and-executable"
challenge "$script_dirS/file-checker/readable-writable-executable"

rm -rf file-checker
```

Reply:
Thanks. Done